src/js/themes/dark.js

```diff
@@ -66,7 +66,25 @@ export default deepFreeze({
     },
   },
   icon: {
-    color: textColor,
+    extend: css`${(props) => {
+      let { color } = props.theme.icon;
+      if (props.color && props.color !== 'plain') {
+        color = colorForName(props.color, props.theme);
+      }
+      if (props.dark === colorIsDark(color)) {
+        color = props.dark ?
```

Comment:
why do you need to check for `props.dark` here?

Reply:
is the user specified a color that would not be visible - for example a dark color on dark background, or light color on light background